jerry-core/parser/js/js-parser-expr.c

```diff
@@ -2823,6 +2951,24 @@ parser_pattern_form_assignment (parser_context_t *context_p, /**< context */
     parser_emit_cbc_ext_forward_branch (context_p, CBC_EXT_DEFAULT_INITIALIZER, &skip_init);
 
     parser_parse_expression (context_p, PARSE_EXPR_NO_COMMA);
+
+    if (name_index != UINT16_MAX)
+    {
+      uint16_t function_literal_index = parser_check_anonymous_function_declaration (context_p);
+
+      if (function_literal_index < PARSER_NAMED_FUNCTION)
```

Comment:
Would it make sense to reorganize the ifs to something like this?:
```c
if (function_literal_index == PARSER_ANONYMOUS_CLASS)
{
}
else if (function_literal_index < PARSER_NAMED_FUNCTION)
{
}
```

Reply:
Good idea.